src/components/year_month.rs

```diff
@@ -48,19 +57,97 @@ impl YearMonth {
         self.iso.year
     }
 
+    /// Returns the padded ISO year string
+    #[inline]
+    #[must_use]
+    pub fn padded_iso_year_string(&self) -> String {
```

Comment:
thought: I'm not entirely sure long term this method should be in our public API.

This method is used for the `toString` implementation on the boa side I believe, which I would much prefer to have the entire `to_string` method implemented on `YearMonth` rather than component parts.

That being said, it's underlying utility method looks pretty fantastic, so I'm not against keeping it and iterating on it until there's a full `to_string` implementation in `temporal_rs`

Reply:
Yep ok, we can come back to this but ill leave it here for now and we can decide going forward  how we want to handle to_string methods